### ISSUE TITLE
Session.set({...}) bug

### ISSUE THREAD

**Issue by Siyfion** [meteor/meteor#3849]
Please read the full description of the issue here: https://forums.meteor.com/t/setting-multiple-session-values-at-once-bug/307 @stubailo asked me to create an issue for it, so here it is!

To reproduce:

``` javascript
Session.set('this.is.a.test', 'some random value');
Session.set({ 'this.is.a.test': '' });
console.log(Session.get('this.is.a.test')); // > 'some random value'
```

**Comment by glasser** [comment]
Hmm, I can't find the PR that added this feature.  But yeah, I have no idea why there is a `if (value)` inside `_setObject`.  Also, the new set syntax is not documented.  @stubailo can you fix these before 1.0.4 RC on Thursday?  Will track in #3814.

**Comment by stubailo** [comment]
@glasser as far as I know, it is documented under the full API documentation for `set`. But will fix that `if (value)` bug.

**Comment by glasser** [comment]
Thanks, looks like you fixed this in 8323213.

It's a shame there are no tests for the new Session.set syntax though.

**Comment by stubailo** [comment]
Yeah that's a good point. I should do those, even if they are just duplicates of the existing tests with the new syntax.
